### DISCUSSION TITLE
Mypy support

### DISCUSSION THREAD

**Discussion by samupl** [django-json-api/django-rest-framework-json-api#1205] [Ideas]
Hey,

Would it be possible to have type hints for this library? Currently when using this library in my project where I wanted to add static type checks, mypy complains:

```
error: Skipping analyzing "rest_framework_json_api.relations": module is installed, but missing library stubs or py.typed marker [import]
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

Is that something this library would be interested in supporting?

**Comment by sliverc** [comment]
Thanks for raising this discussion.

Having mypy support could be something useful for DJA. However, as outlined in our [3rd party policy](https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#third-party-packages) we want to avoid adding too many features to DJA directly. Secondly Django REST framework also seems to have its mypy support in an [external library](https://pypi.org/project/djangorestframework-stubs/).

Therefore, I would prefer seeing such a stubs support in a separate library as well. If anyone is willing to work on this, best to create a repository and raise it here so other people who are interested can join in the effort as well.
